### PR TITLE
DT-1581 initial library

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  publications {
+    create<MavenPublication>("maven") {
+      groupId = "uk.gov.justice.hmpps"
+      artifactId = "hmpps-spring-boot-sqs"
+      version = "0.0.1-SNAPSHOT"
+
+      from(components["java"])
+    }
+  }
+}
```
Comment (mikehalmamoj) [inline build.gradle.kts:58]: Temporarily allow publishing locally for testing until we work out how and where to publish permanently.

File: src/test/kotlin/uk/gov/justice/hmpps/sqs/SqsQueueAdminServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      @Test
+      fun `Deletes message from the dlq`() {
+        queueAdminService.transferAllMessages(TransferMessagesRequest(dlqSqs, "dlqUrl", queueSqs, "queueUrl"))
+
+        val captor = argumentCaptor<DeleteMessageRequest>()
+        verify(dlqSqs, times(2)).deleteMessage(captor.capture())
+
+        assertThat(captor.firstValue.receiptHandle).isEqualTo("message-1-receipt-handle")
+        assertThat(captor.secondValue.receiptHandle).isEqualTo("message-2-receipt-handle")
+      }
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/hmpps/sqs/SqsQueueAdminServiceTest.kt:161]: Using an argument captor to workaround this issue: https://github.com/mockito/mockito-kotlin/issues/288

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+group = "uk.gov.justice.hmpps"
+version = "0.0.1-SNAPSHOT"
+
+dependencies {
+  implementation("org.springframework.boot:spring-boot-starter")
+  implementation("com.amazonaws:amazon-sqs-java-messaging-lib:1.0.8")
+
+  implementation("org.springframework.boot:spring-boot-starter-test")
```
Comment (petergphillips) [inline build.gradle.kts:23]: this looks odd - you've got a test library as `implementation`?

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  testImplementation("org.assertj:assertj-core:3.18.1")
```
Comment (petergphillips) [inline build.gradle.kts:24]: think these two are automatically included anyway by `spring-boot-starter-test`

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**